FinSearchUnified/BusinessLogic/ExportService.php

```diff
@@ -0,0 +1,254 @@
+
+    /**
+     * @var ExportErrorInformation[][]
+     */
+    protected $errors = [];
+
+    public function __construct($shopkey, Category $baseCategory)
+    {
+        $this->customerRepository = Shopware()->Container()->get('models')->getRepository(Customer::class);
+        $this->articleRepository = Shopware()->Container()->get('models')->getRepository(Article::class);
+        $this->findologicArticleFactory = Shopware()->Container()->get('fin_search_unified.article_model_factory');
+        $this->shopkey = $shopkey;
+        $this->baseCategory = $baseCategory;
+        $this->allUserGroups = $this->customerRepository->getCustomerGroupsQuery()->getResult();
+    }
+
+    /**
+     * @return int
+     */
+    public function fetchTotalProductCount()
+    {
+        $countQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('count(articles.id)')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        return intval($countQuery->getQuery()->getScalarResult()[0][1]);
+    }
+
+    /**
+     * @param int $start
+     * @param int $count
+     *
+     * @return Article[]
+     */
+    public function fetchAllProducts($start, $count)
+    {
+        $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        if ($count > 0) {
+            $articlesQuery->setMaxResults($count)->setFirstResult($start);
+        }
+
+        return $articlesQuery->getQuery()->execute();
+    }
+
+    /**
+     * @param string $productId
+     *
+     * @return Article[]
+     */
+    public function fetchProductsById($productId)
+    {
+         $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.id = :productId')
+            ->orWhere('articles.supplier = :productId')
+            ->setParameter('productId', $productId);
+
+        $shopwareArticles = $articlesQuery->getQuery()->execute();
+
+        if (count($shopwareArticles) === 0) {
+            $errorInformation = new ExportErrorInformation($productId);
+            $errorInformation->errors[] = 'No article found with given ID';
+
+            $this->errors['general'][] = $errorInformation;
+            return null;
+        }
+
+        return $shopwareArticles;
+    }
+
+    /**
+     * @param array $shopwareArticles
+     * @param false $logErrors
+     *
+     * @return Item[]
+     */
+    public function generateFindologicProducts($shopwareArticles, $logErrors = false)
+    {
+        $findologicArticles = [];
+
+        /** @var Article $article */
+        foreach ($shopwareArticles as $article) {
+            try {
+                $findologicArticle = self::getFindologicArticle($article, $logErrors);
```

Comment:
`self::getFindologicArticle` is not a static method, so call it like a normal method with `$this->getFindologicArticle`.